### DISCUSSION TITLE
async RSC Streaming Content after </html> - Abuse of standards?

### DISCUSSION THREAD

**Discussion by matt-kruse** [vercel/next.js#51291] [App Router] [answered]
When an async RSC uses Suspense, the streamed content (div, script) gets inserted after the &lt;/html&gt; closing tag. This is technically an abuse of standards, as there shouldn't be content there. Is it dangerous to bet on browsers being non-standards-compliant for core functionality?
Further, I'm trying to capture these insertions for logging and debugging, and I can't seem to catch them. Neither MutationObserver nor DOMNodeInserted get fired when content is inserted after &lt;/html&gt;.
This seems troubling to me. Isn't there a better way that doesn't break standards?

**Comment by icyJoseph** [comment, marked as answer]
Which version of Next.js do you have? IIRC they fixed the script tags after HTML closing a couple of versions back.

**Reply by matt-kruse** [reply]
13.4.? has this behavior. I will update to latest canary and see if it's still that way.

**Reply by icyJoseph** [reply]
I mean, maybe we are talking about two different things:

- https://github.com/vercel/next.js/issues/48372

But it used to be that the HTML contained script tags after the closing HTML, but it doesn't anymore.

**Reply by matt-kruse** [reply]
I updated to 13.4.5 and now the tags seem to be inside </body>. Nice!

**Reply by icyJoseph** [reply]
Please don't forget to mark answered 🎉
